crates/rspack_plugin_split_chunks_new/src/common.rs

```diff
@@ -54,7 +56,7 @@ pub fn create_module_filter_from_rspack_regex(re: rspack_regex::RspackRegex) ->
   })
 }
 
-pub fn create_module_filter(re: Option<String>) -> ModuleFilter {
+pub fn create_module_filter(re: Option<Either<JsString, JsRegExp>>) -> ModuleFilter {
```

Comment:
Currently, Rspack only supports `JsRegExp`, so `JsRegExp`  is converted to `RspackRegex`, which we used to create a [closure](https://github.com/web-infra-dev/rspack/blob/5b145f69eadd23f3e79df174bf15743509fe1ded/crates/rspack_plugin_split_chunks_new/src/common.rs#L51-L58) for porting this part code: https://github.com/webpack/webpack/blob/f921a5baca9c7d200eee8204fafaaaf36b7c1451/lib/optimize/SplitChunksPlugin.js#L530-L533.

Now, we want Rspack to support `String` and the corresponding code of Webpack is https://github.com/webpack/webpack/blob/f921a5baca9c7d200eee8204fafaaaf36b7c1451/lib/optimize/SplitChunksPlugin.js#L526-L529.

So, you need to rewrite the `create_module_filter` function.

The behavior should be
- If the passing value is a `JsRegExp` return a closure does [this](https://github.com/webpack/webpack/blob/f921a5baca9c7d200eee8204fafaaaf36b7c1451/lib/optimize/SplitChunksPlugin.js#L530-L533).
  - Actually, we already implement it in the current codebase.
- If the passing value is `String`, return a closure does [this](https://github.com/webpack/webpack/blob/f921a5baca9c7d200eee8204fafaaaf36b7c1451/lib/optimize/SplitChunksPlugin.js#L526-L529).